op-service/dial/l2_provider.go

```diff
@@ -0,0 +1,69 @@
+package dial
+
+import (
+	"context"
+
+	"github.com/ethereum-optimism/optimism/op-service/sources"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+)
+
+type L2EndpointProvider interface {
+	EthClient(ctx context.Context) (*ethclient.Client, error)        // op-geth
+	RollupClient(ctx context.Context) (*sources.RollupClient, error) // op-node
+	Close()
+}
```

Comment:
I'm wondering if we might want to create two provider interfaces, one that provides both, a rollup and eth client, and another one that only provides the rollup client. This would be safer, we wouldn't ever create an unsafe static provider that returns `nil` ethclients.

```suggestion
type RollupProvider interface {
	RollupClient(ctx context.Context) (*sources.RollupClient, error) // op-node
	Close()
}

type L2EndpointProvider interface {
	RollupProvider
	EthClient(ctx context.Context) (*ethclient.Client, error)        // op-geth
}
```

You can then also create two different static provider impls, and have the one contain the other one.